### PR TITLE
Search for relevant container ID when provided with a short version of it

### LINKED ISSUE #37: Have DE auto complete partial docker ID
ie use `0ddf71064` for 
`0ddf710643b47e581dbce3e17677c9a557b82943bacee2dfbc19f178a99ad374`

### PR DESCRIPTION
**Pull request by rgayon**
Fixes #37

### CODE REVIEW

File: docker_explorer/de.py [hunk 1]
```diff
@@ -139,15 +139,49 @@ def ParseOptions(self, options):
...
+    possible_cids = []
+    for container_dirs in sorted(os.listdir(containers_dir)):
+      possible_cid = os.path.basename(container_dirs)
+      if possible_cid.startswith(short_id):
+        possible_cids.append(possible_cid)
+
+    possible_cids_len = len(possible_cids)
+    if possible_cids_len == 0:
+      raise errors.DockerExplorerError(
+          'Could not find any container whose ID starts with "{0}"'.format(
```
Comment (tomchop) [inline docker_explorer/de.py]: I would rephrase to `Could not find any container IDs starting with {0:s}`

File: docker_explorer/de.py [hunk 2]
```diff
@@ -139,15 +139,49 @@ def ParseOptions(self, options):
...
+
+    containers_dir = os.path.join(self.docker_directory, 'containers')
+    possible_cids = []
+    for container_dirs in sorted(os.listdir(containers_dir)):
+      possible_cid = os.path.basename(container_dirs)
+      if possible_cid.startswith(short_id):
+        possible_cids.append(possible_cid)
+
+    possible_cids_len = len(possible_cids)
+    if possible_cids_len == 0:
```
Comment (tomchop) [inline docker_explorer/de.py]: Can be replaced by `if not possible_cids`

File: docker_explorer/de.py [hunk 3]
```diff
@@ -139,15 +139,49 @@ def ParseOptions(self, options):
...
+      raise errors.DockerExplorerError(
+          'Could not find any container whose ID starts with "{0}"'.format(
+              short_id))
+    elif possible_cids_len > 1:
```
Comment (tomchop) [inline docker_explorer/de.py]: Some people would argue the "else" is not necessary since you `raise` in the other branch.

File: docker_explorer/de.py [hunk 4]
```diff
@@ -139,15 +139,49 @@ def ParseOptions(self, options):
...
+      raise errors.DockerExplorerError(
+          'Too many containers whose ID starts with "{0}" ({1:d})'.format(
```
Comment (tomchop) [inline docker_explorer/de.py]: Would rephrase to `Too many container IDs starting with "{0:s}"`. Also it may be nice to list them?

File: docker_explorer/de.py [hunk 5]
```diff
@@ -139,15 +139,49 @@ def ParseOptions(self, options):
...
+  def GetContainer(self, container_id_part):
+    """Returns a Container object given the first characters of a container_id.
+
+    Args:
+      container_id_part (str): the first characters of a container ID.
 
     Returns:
       container.Container: the container object.
     """
+    container_id = self._GetFullContainerID(container_id_part)
```
Comment (tomchop) [inline docker_explorer/de.py]: You don't seem to be catching any exceptions raised by `_GetFullContainerID`

File: docker_explorer/de.py (hunk 2 above)
Comment (rgayon) [reply docker_explorer/de.py]: I believe the current code is more explicit (especially with the other check that it's more than 1)

File: docker_explorer/de.py (hunk 3 above)
Comment (rgayon) [reply docker_explorer/de.py]: It does look cleaner without =)

File: docker_explorer/de.py (hunk 4 above)
Comment (rgayon) [reply docker_explorer/de.py]: Rephrased

File: docker_explorer/de.py (hunk 5 above)
Comment (rgayon) [reply docker_explorer/de.py]: Well I do want the script to fail terribly here if we didn't find one and exactly one container ID

Comment (tomchop) [reply docker_explorer/de.py]: I guess this comes down to choosing between catching and displaying an error to the user vs. displaying a stacktrace.

Comment (rgayon) [reply docker_explorer/de.py]: "code is hard to write, it should be hard to use"
On a more serious note, if I catch i'll just display the error message and sys.exit(1), or return None and make a lot of checks "if container_obj" after calls to GetContainer(str).
I think having the stacktrace shown to the user is fine

Comment (tomchop) [reply docker_explorer/de.py]: Fair enough.

File: docker_explorer/de.py (hunk 2 above)
Comment (tomchop) [reply docker_explorer/de.py]: Ack
